EssentialFeed/EssentialFeedTests/Image Comments/API/ImageCommentsMapperTests.swift

```
@@ -0,0 +1,87 @@
+//
+// Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class ImageCommentsMapperTests: XCTestCase {
+	func test_map_throwsErrorOnNon2xxHTTPResponse() throws {
+		let json = makeItemsJSON([])
+		let non2xxSamples = [199, 300, 400, 500]
+
+		assert(forEach: non2xxSamples, with: json, toGetResult: .failure(anyNSError()))
+	}
```

This test is misleading:

1. The test reads as checking if the `map` function `throws` an error, but the error gets ignored inside the `assert` function and no `XCTAssertThrowsError` are used.
2. The `assert` function is being called to check against `anyNSError()`, but the `map` function throws errors of type `ImageCommentsMapper.Error.invalidData`, so the assertion is wrong.

---

Not occurring anymore. The helper function now just test if an exception is thrown by the map function.